packages/amplify-category-api/src/provider-utils/awscloudformation/base-api-stack.ts

```diff
@@ -15,7 +15,7 @@ import { NETWORK_STACK_LOGICAL_ID } from '../../category-constants';
 import Container from './docker-compose/ecs-objects/container';
 import { GitHubSourceActionInfo, PipelineWithAwaiter } from './pipeline-with-awaiter';
 
-const PIPELINE_AWAITER_ZIP = 'custom-resource-pipeline-awaiter.zip';
+const PIPELINE_AWAITER_ZIP = 'custom-resource-pipeline-awaiter-18.zip';
```

Comment:
Does this need to be parameterized? Under what conditions would this number need to change in the future? What is 18 referring to?

Reply:
This name would need to change anytime [this file](https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-provider-awscloudformation/resources/custom-resource-pipeline-awaiter.zip) is modified in the CLI repo. 18 is referring the node runtime version. The zip file was updated to be compatible with Node 18.

I think parametrization would be possible but I think would require large changes to the API category and the CLI.

I chose the option to simply update the name (which will update the S3 key) because it would be the fastest method to release a fix for this while still being effective.

Reply:
This is the name of the key as stored in S3 and this naming convention is shared b/w CLI and API plugins [reference](https://github.com/aws-amplify/amplify-cli/blob/5ec5d4d6678a345346aad2f4a4fe5ba5b518b08c/packages/amplify-provider-awscloudformation/src/resource-package/resource-packager.ts#L60) with the assumption on CLI side that the cloudformation provider would upload the lambda code in the deployment bucket using that key and we use it as part of the code pipeline state machine in our REST API deployments.

Reply:
Node 18 is EOL in 9 months. How do we prevent missing this 9 months from now? https://github.com/nodejs/Release

Reply:
This key doesn't require a change for every Lambda runtime update but when the underlying code is not compatible with older versions which coincidentally the case from 16.x -> 18.x runtime since lambda changed it's default support of AWS SDK version from 2.x to 3.x as part of this runtime update [reference](https://aws.amazon.com/blogs/compute/node-js-18-x-runtime-now-available-in-aws-lambda/). We should have an E2E test added to the migration tests that CLI has so that we can catch such cases where a runtime update requires code change. We could do this as a follow-up after the fix is verified to work for this case.